v3/generator/operations/operations.go

```diff
@@ -258,6 +266,95 @@ func renderRequestParametersSchema(name string, op *v3.Operation) ([]byte, error
 	return output.Bytes(), nil
 }
 
+const gettableTemplate = `
+// Get{{ .TypeName }} attempts to find an {{ .TypeName }} by name or ID.
+func (l {{ .ListTypeName }}) Get{{ .TypeName }}(nameOrID string) ({{ .TypeName }}, error) {
+	for i, elem := range l.{{ .ListFieldName }} {
+		if elem.Name == nameOrID || elem.ID.String() == nameOrID {
+			return l.{{ .ListFieldName }}[i], nil
+		}
+	}
+
+	return {{ .TypeName }}{}, ErrNotFound
+}
+`
+
+type Gettable struct {
+	TypeName      string
+	ListTypeName  string
+	ListFieldName string
+}
+
+// renderGettable renders a gettable method on listable resource.
+// this gettable method get the resource by name or id if available.
+// returns nil on non listable resources.
+func renderGettable(funcName string, s *base.SchemaProxy) ([]byte, error) {
+	sc, err := s.BuildSchema()
+	if err != nil {
+		return nil, err
+	}
+	schemas.InferType(sc)
+
+	// Check if listable response.
+	if !strings.HasPrefix(strings.ToLower(funcName), "list") {
+		return nil, nil
+	}
+
+	if sc.Type[0] != "object" {
+		return nil, nil
+	}
+
+	for propName, propSc := range sc.Properties {
+		prop, err := propSc.BuildSchema()
+		if err != nil {
+			return nil, err
+		}
+		schemas.InferType(prop)
+
+		if prop.Type[0] != "array" {
+			continue
+		}
+
+		if prop.Items == nil {
```

Comment:
you can also check if `len(prop.Items) > 0` because len returns 0 if the argument is `nil`. That way both cases are covered.

Reply:
I'm aware of it, but in this case it's not a slice, so `len()` can not apply